test/tests/transact.ts

```diff
@@ -192,55 +135,133 @@ suite('transact', function () {
                 assetValidTransactResponse(result)
             })
         })
+        suite('invalid', function () {
+            test('no abi for contract', async function () {
+                const {action, session} = await mockData()
+                const data = Serializer.objectify(action)
+                data.account = ''
+                session.transact({action: data}).catch((error) => {
+                    assert.equal(error.message, 'No data for /v1/chain/get_abi')
+                })
+            })
+        })
     })
-    suite('transact options', function () {
+    suite('options', async function () {
         suite('allowModify', function () {
-            test('undefined', async function () {
-                const result = await session.transact({action})
+            test('default: true', async function () {
+                const {action, session} = await mockData()
+                const result = await session.transact(
+                    {action},
+                    {
+                        allowModify: true,
+                        transactPlugins: [new MockTransactResourceProviderPlugin()],
```

Comment:
The `TransactPlugins` can also be overridden on an individual `.transact` call against a session.